### DISCUSSION TITLE
ItemsRepeater list of whats loaded

### DISCUSSION THREAD

**Discussion by SirZeeno** [AvaloniaUI/Avalonia#16308] [Q&A] [answered]
I was wondering if there is a list of all the items that the itemsRepeater has loaded that I might be able to read from before I create a function that checks which items are visible on the screen. I am trying to get each of the items to load something when they are loaded, but I can't have them all load separately due to database connection limitations, so I am trying to see what's loaded so I can manually pool them together to load what they need to load.

**Comment by timunie** [comment]
Hmmm 🤔 

VisualTreeHelper could be handy. ItemsControl also holds a list of materialized items but dunno for ItemsRepeater. 

Anyways I believe this belongs into the ViewModel, not the UI layer...

**Comment by thevortexcloud** [comment, marked as answer]
> I am trying to get each of the items to load something when they are loaded,

Can't you just lazy load things? Otherwise ItemsRepeater has this event.

https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.itemsrepeater.elementprepared?view=winui-2.2

FYI: ItemsRepeater in Avalonia is just a port of the WinUI control. So all the docs are valid for Avalonia as well.

**Comment by maxkatz6** [comment]
> before I create a function that checks which items are visible on the screen

Normally, I would recommend using something like `item.GetTransformedBounds()` and checking the results comparing to the parent or items repeater boundaries. If it's inside of the parent - it's visible, if outside - not.

> I am trying to get each of the items to load something when they are loaded,

But when you need to do something when item is loaded, then yes - element prepared event, as @thevortexcloud recommended, would work better for you.
